Octokit/Models/Request/NewRepositoryHook.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Octokit.Internal;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class NewRepositoryHook
+    {
+        public NewRepositoryHook(string name, dynamic config)
+        {
+            Name = name;
+            Config = config;
+        }
+
+        [Parameter(Key = "name")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Is a key value structure determined by the web hook being created
+        /// </summary>
+        [Parameter(Key = "config")]
+        public dynamic Config { get; set; }
```

Comment:
Changed from the dynamic type to Dictionary<string, string>.
I had to place the config in the constructor of the 'EditRepositoryHook' with a default argumentless constructor also.

The reason being the code analysis requires the dictionary not to have a public 'set' and I needed the Config property to be default 'null', because else the edit call would overwrite the config settings even if the user did not provide any.